test/core/transport/chaotic_good/client_transport_error_test.cc

```diff
@@ -0,0 +1,427 @@
+  event_engine_->TickUntilIdle();
+  event_engine_->UnsetGlobalHooks();
+}
+
+TEST_F(ClientTransportTest, AddMultipleStreamWithWriteFailed) {
+  // Mock write failed at first stream and second stream's write will fail too.
+  EXPECT_CALL(control_endpoint_, Write)
+      .Times(1)
+      .WillRepeatedly(
+          WithArgs<0>([](absl::AnyInvocable<void(absl::Status)> on_write) {
+            on_write(absl::InternalError("control endpoint write failed."));
+            return false;
+          }));
+  EXPECT_CALL(data_endpoint_, Write)
+      .Times(1)
+      .WillRepeatedly(
+          WithArgs<0>([](absl::AnyInvocable<void(absl::Status)> on_write) {
+            on_write(absl::InternalError("data endpoint write failed."));
+            return false;
+          }));
+  EXPECT_CALL(control_endpoint_, Read)
+      .InSequence(control_endpoint_sequence_)
+      .WillOnce(Return(false));
+  InitialClientTransport();
+  ClientMetadataHandle first_stream_md;
+  ClientMetadataHandle second_stream_md;
+  auto first_stream_args =
+      CallArgs{std::move(first_stream_md),
+               ClientInitialMetadataOutstandingToken::Empty(),
+               nullptr,
+               &pipe_server_intial_metadata_.sender,
+               &pipe_client_to_server_messages_.receiver,
+               &pipe_server_to_client_messages_.sender};
+  auto second_stream_args =
+      CallArgs{std::move(second_stream_md),
+               ClientInitialMetadataOutstandingToken::Empty(),
+               nullptr,
+               &pipe_server_intial_metadata_second_.sender,
+               &pipe_client_to_server_messages_second_.receiver,
+               &pipe_server_to_client_messages_second_.sender};
+  StrictMock<MockFunction<void(absl::Status)>> on_done;
+  EXPECT_CALL(on_done, Call(absl::OkStatus()));
+  auto activity = MakeActivity(
+      Seq(
+          // Concurrently: write and read messages from client transport.
+          Join(
+              // Add first stream with call_args into client transport.
+              // Expect return trailers "grpc-status:unavailable".
+              AddStream(std::move(first_stream_args)),
+              // Send messages to first stream's
+              // call_args.client_to_server_messages pipe.
+              SendClientToServerMessages(pipe_client_to_server_messages_, 1)),
+          Join(
+              // Add second stream with call_args into client transport.
+              // Expect return trailers "grpc-status:unavailable".
+              AddStream(std::move(second_stream_args)),
+              // Send messages to second stream's
+              // call_args.client_to_server_messages pipe.
+              SendClientToServerMessages(pipe_client_to_server_messages_second_,
+                                         1)),
+          // Once complete, verify successful sending and the received value.
+          [](const std::tuple<ServerMetadataHandle, absl::Status>& ret) {
```

Comment:
This only checks the result of the second join. Could you add a similar blob after the first join in line 332 ?

Reply:
Done, thanks for the suggestion.